.gitlab-ci.yml

```diff
@@ -1838,3 +1838,77 @@ chef-release:
     - cp -r deployments/chef /tmp/cookbooks/splunk_otel_collector
   script:
     - knife supermarket share -o /tmp/cookbooks splunk_otel_collector
+
+.trigger-dotnet-instrumentation-deployer:
+  only:
+    variables:
+      - $CI_COMMIT_TAG =~ /^dotnet-instrumentation-deployer-ta-v[0-9]+\.[0-9]+\.[0-9]+$/
+  except:
+    - schedules
+
+build-dotnet-instrumentation-deployer:
+  extends: .trigger-dotnet-instrumentation-deployer
+  image: '${DOCKER_HUB_REPO}/golang:${GO_VERSION}'
+  stage: build-dotnet-deployer
+  before_script:
+    - cd packaging/dotnet-instr-deployer-add-on
+  script:
+    - make build
+  artifacts:
+    paths:
+      - packaging/dotnet-instr-deployer-add-on/assets/windows_x86_64/bin/*
+
+sign-dotnet-instrumentation-deployer-exe:
+  extends:
+    - .trigger-filter
+    - .submit-signing-request
+  stage: sign-dotnet-deployer
+  retry: 2
+  needs:
+    - build-dotnet-instrumentation-deployer
+  variables:
+    ARTIFACT: packaging/dotnet-instr-deployer-add-on/assets/windows_x86_64/bin/splunk_otel_dotnet_deployer.exe
+    SIGN_TYPE: WIN
+    DOWNLOAD_DIR: packaging/dotnet-instr-deployer-add-on/assets/windows_x86_64/bin
+  artifacts:
+    paths:
+      - packaging/dotnet-instr-deployer-add-on/assets/**/*
+
+pack-dotnet-instrumentation-deployer:
+  extends: .trigger-dotnet-instrumentation-deployer
+  image: '${DOCKER_HUB_REPO}/golang:${GO_VERSION}'
+  stage: pack-dotnet-deployer
+  needs:
+    - sign-dotnet-instrumentation-deployer-exe
+  before_script:
+    - cd packaging/dotnet-instr-deployer-add-on
+  script:
+    - make pack
+  artifacts:
+    paths:
+      - packaging/dotnet-instr-deployer-add-on/out/distribution/splunk_otel_dotnet_deployer.tgz
+
+dotnet-instrumentation-deployer-release:
+  extends: .trigger-dotnet-instrumentation-deployer
+  image: '${DOCKER_HUB_REPO}/golang:${GO_VERSION}'
+  stage: release-dotnet-deployer
+  needs:
+    - pack-dotnet-instrumentation-deployer
+  before_script:
+    - cd packaging/dotnet-instr-deployer-add-on
+  dependencies:
+    - dotnet-instrumentation-deployer-builder
+  script:
+    - pushd out/distribution/ && shasum -a 256 * > checksums.txt && popd
+    - |
+      set -e
+      if [[ "${CI_COMMIT_TAG:-}" =~ ^dotnet-instrumentation-deployer-ta-v([0-9]+\.[0-9]+\.[0-9]+[^ ]*) ]]; then
+        VERSION=${BASH_REMATCH[1]}
+        VERSION_CHANGELOG="$(awk -v version="$VERSION" '/^## / { if (p) { exit }; if ($2 == version) { p=1; next } } p && NF' CHANGELOG.md)"
+        SPLUNK_OTEL_DOTNET_VERSION=$(grep -oP 'version = \K.*' assets/windows_x86_64/bin/Splunk.OTel.DotNet.psm1 | tr -d '"')
```

Comment:
This is a bit fragile but since it is created as draft we can always fix before publishing it.